### PR TITLE
Disallow GHC <8.0 for Cabal{,-syntax} & cabal-install-solver

### LINKED ISSUE #8715: Cabal-syntax allows GHC-7.8 and GHC-7.10 buildplans
And I checked, `Cabal-syntax-3.9.0.0` as in master **has** build plans with GHC-7.8 and GHC-7.10, but these fail to build

Please fix the metadata.

### PR DESCRIPTION
**Pull request by Kleidukos**
(hopefully) closes #8715

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

### CODE REVIEW

**Review by ffaf1 — APPROVED** [review]

**Comment by ulysses4ever** [comment]
We recently dropped 8.2 from CI (leave alone 8.0): are we sure that all this stuff builds with those GHCs? Maybe we should raise the bar even higher, so to 8.4? I think it doesn't violate the support policy because that's what we checked when dropping 8.2...

**Comment by Mikolaj** [comment]
> We recently dropped 8.2 from CI (leave alone 8.0): are we sure that all this stuff builds with those GHCs? Maybe we should raise the bar even higher, so to 8.4? I think it doesn't violate the support policy because that's what we checked when dropping 8.2...

If they still build, no harm leaving the bounds permissive. Let's wait until somebody actually tries it and complains.

**Review by Mikolaj — APPROVED** [review]
LGTM.

**Comment by Mikolaj** [comment]
@mergify backport 3.10

File: cabal-testsuite/cabal-testsuite.cabal [hunk 1]
```diff
@@ -26,7 +26,7 @@ common shared
   default-language: Haskell2010
 
   build-depends:
-    , base >= 4.6 && <4.18
```
Comment (andreasabel) [inline cabal-testsuite/cabal-testsuite.cabal:29]: Shouldn't we conform this to `<5` as in the other cases?
Especially in the testsuite we do _not_ need _more_ restrictive bounds than in the shipped products.
(Not really topic of this PR, but while we are at it...)

**Pull request merged**